src/NetTopologySuite/Triangulate/QuadEdge/Vertex.cs

```diff
@@ -30,9 +30,41 @@ public class Vertex : IEquatable<Vertex>
         private const int ORIGIN = 5;
         private const int DESTINATION = 6;
 
+
+        
+        private int id; // ID of the vertex	
         private readonly Coordinate _p;
+        private bool border;
```

Comment:
Per the summary documentation on this class, the intent is to make your own subclass to add these properties, instead of forcing everyone to pay for them.

Unfortunately there's no convenient way to do this using `ConformingDelaunayTriangulationBuilder`, you have to use `ConformingDelaunayTriangulator` more directly.